python/pylib/pose.py

```diff
@@ -20,912 +16,1317 @@ def quatInit():
-    q_n2b = mul_Quat_Quat_Array(q_e2b, q_n2e)
-    eul_n2b = quat2eulerArray(quat)
+    q_n2e = euler2quat(np.array([pi, 0.0, 0.5*pi]))
+    q_e2b = euler2quat(eul_e2b)
+    q_n2b = mul_Quat_Quat(q_e2b, q_n2e)
+    eul_n2b = quat2euler(q_n2b)
     return eul_n2b
 
 
-#  * NED to Euler_t
+#  Compute body atitude as Euler angles (pitch and heading with zero roll)
+#  given body X unit vector NED cordinates
+#  TODO: this function needs a better name
 def nedEuler(ned):
-    euler = np.zeros(np.shape(ned))
-    euler[:,2] = arctan2( ned[:,1], ned[:,0] )
-    euler[:,1] = arctan2( -ned[:,2], np.sqrt( ned[:,0]**2 + ned[:,1]**2 ) )
-    euler[:,0] = 0
-    return euler
-
-
-#  * Euler_t to NED
-def eulerNed( e ):
-    ned = np.zeros(3)
-    v = r_[ 1., 0., 0.]
-    vectorRotateBodyToInertial( v, e, ned )
-    return ned
-
-
-# Rotate theta eulers from body to inertial frame by ins eulers, in order: phi, theta, psi
-def eulerRotateBodyToInertial2( e, rot ):
-    eResult = np.zeros(np.shape(e))
+    if len(np.shape(ned)) == 1:
+        ned = np.expand_dims(ned, axis=0)
+        array = 0
+    else:
+        array = 1
 
-    Ai = eulerDCM(rot)                  # use estimate
+    euler = np.empty(shape=(np.shape(ned)[0], 3))
+    euler[:,2] = arctan2(ned[:,1], ned[:,0])
+    euler[:,1] = arctan2(-ned[:,2], np.sqrt(ned[:,0]**2 + ned[:,1]**2))
+    euler[:,0] = 0.0
 
-    for i in range(0,np.shape(eResult)[0]):
-        # Create DCMs (rotation matrices)
-        At = eulerDCM(e[i,:])
-         
-        # Apply INS Rotation to Desired Target vector
-        AiAt    = dot(Ai, At)               # Apply rotation
-        eResult[i,:] = DCMeuler(AiAt)            # Pull out new eulers
+    if array == 0:
+        euler = np.squeeze(euler)
+    return euler
 
-    return eResult
 
-# Rotate theta eulers from inertial to body frame by ins eulers, in order: psi, theta, phi
-def eulerRotateInertialToBody2( e, rot ):
-    eResult = np.zeros(np.shape(e))
+#  Compute coordinates of a body X unit vector in NED
+#  given body attitude in NED as Euler angles
+#  TODO: this function needs a better name
+def eulerNed(euler):
+    if len(np.shape(euler)) == 1:
+        euler = np.expand_dims(euler, axis=0)
+        array = 0
+    else:
+        array = 1
+    e0 = np.array([1.0, 0.0, 0.0])
+    q  = euler2quat(euler)
+    e0n = quatRot(q, e0)
+    if array == 0:
+        e0n = np.squeeze(e0n)
+    return e0n
 
-    Ai = eulerDCM(rot)                  # use estimate
 
-    for i in range(0,np.shape(eResult)[0]):
-        # Create DCMs (rotation matrices)
-        At = eulerDCM(e[i,:])
-         
-        # Apply INS Rotation to Desired Target vector
-        AiAt    = dot(Ai.T, At)             # Apply rotation
-        eResult[i,:] = DCMeuler(AiAt)            # Pull out new eulers
+#  Compute body atitude quaternion assuming zero roll and
+#  given body X unit vector NED cordinates
+#  TODO: this function needs a better name
+def nedQuat(v):
```

Comment:
This function could benefit from a better name. I added description based on what I think it does (no prior description)